### PR TITLE
Fixing issues introduced by upgrading to ensembl-utils

### PR DESCRIPTION
**Pull request by ens-sb**
## Description

We need to upgrade to using ensembl-utils in order to be able to use the new modular environment.

**Related JIRA tickets:**
- ENSCOMPARASW-7625

## Overview of changes

Miscalleneous changes related to updates in pytest fixtures and other issues.

## Testing
Testing was done by Travis. There are remaining pytest and pylint issues which will be addressed in [ENSCOMPARASW-7667](https://www.ebi.ac.uk/panda/jira/browse/ENSCOMPARASW-7667)

## Notes
Many thanks to @JAlvarezJarreta for the generous contributions to this PR.

---

For code reviewers: [code review SOP](https://www.ebi.ac.uk/seqdb/confluence/display/EnsCom/Code+review+SOP)

### CODE REVIEW

File: src/python/tests/test_filesys.py [hunk 1]
```diff
@@ -65,10 +66,10 @@ class TestDirCmp(BaseTestFilesys):
     """Tests :class:`DirCmp` class."""
 
     @pytest.mark.dependency(name='test_init', scope='class')
-    def test_init(self, tmp_dir: Path) -> None:
+    def test_init(self, tmp_path: Path) -> None:
```
Comment (twalsh-ebi) [inline src/python/tests/test_filesys.py:69]: This fixture is no longer used in this method:
```suggestion
    def test_init(self) -> None:
```

File: src/python/tests/test_repair_mlss_tags.py [hunk 2]
```diff
@@ -39,21 +40,21 @@ class TestRepairMLSSTags:
 
     """
 
-    dbc = None  # type: DBConnection
+    dbc: DBConnection | None = None
```
Comment (twalsh-ebi) [inline src/python/tests/test_repair_mlss_tags.py:43]:
```suggestion
    dbc: DBConnection = None
```

Comment (JAlvarezJarreta) [reply src/python/tests/test_repair_mlss_tags.py:43]: Whilst the code will still work, `mypy` will complain if you say `dbc` type does not allow `None` values.

File: src/python/tests/test_repair_mlss_tags.py [hunk 3]
```diff
@@ -82,31 +83,37 @@ def setup(self, db: UnitTestDB) -> None:
...
+            #         "INSERT INTO method_link_species_set_tag VALUES (404, 'msa_mlss_id', 1)"
+            #     ],
+            #     set([
+            #         "Repaired MLSS tag 'msa_mlss_id' for MLSS id '5'",
+            #         "Added missing MLSS tag 'msa_mlss_id' for MLSS id '50001'",
+            #         "Deleted unexpected MLSS tag 'msa_mlss_id' for MLSS id '404'"
+            #     ]),
+            #     {5: 4, 7: 6, 9: 8, 50001: 4, 50002: 6, 50003: 8},
+            #     does_not_raise()
+            # ),
```
[the remark this answers is not in the document]
Comment (JAlvarezJarreta) [reply src/python/tests/test_repair_mlss_tags.py:102]: This is my fault, not sure how my comment during testing got there. Sorry! 🤦

File: src/python/tests/test_repair_mlss_tags.py (hunk 2 above)
Comment (twalsh-ebi) [reply src/python/tests/test_repair_mlss_tags.py:43]: @JAlvarezJarreta, `mypy` didn't complain when I tried this suggestion locally, and it is currently complaining as a result of this type annotation. @ens-sb, why not try my suggestion? If `mypy` complains, I'd be happy to make an alternative suggestion.

**Review by twalsh-ebi — COMMENTED** [review]
🙂

File: src/python/tests/test_repair_mlss_tags.py (hunk 2 above)
Comment (ens-sb) [reply src/python/tests/test_repair_mlss_tags.py:43]: @twalsh-ebi @JAlvarezJarreta Maybe differing mypy versions cause this discrepancy? In any case I will try the susggestion and backtrack if fails.

File: src/python/tests/test_repair_mlss_tags.py [hunk 4]
```diff
@@ -82,31 +83,37 @@ def setup(self, db: UnitTestDB) -> None:
...
+                     "INSERT INTO method_link_species_set_tag VALUES (404, 'msa_mlss_id', 1)"
+                 ],
+                 set([
+                     "Repaired MLSS tag 'msa_mlss_id' for MLSS id '5'",
+                     "Added missing MLSS tag 'msa_mlss_id' for MLSS id '50001'",
+                     "Deleted unexpected MLSS tag 'msa_mlss_id' for MLSS id '404'"
+                 ]),
+                 {5: 4, 7: 6, 9: 8, 50001: 4, 50002: 6, 50003: 8},
+                 does_not_raise()
+             ),
```
Comment (twalsh-ebi) [inline src/python/tests/test_repair_mlss_tags.py:102]: My apologies, I think I missed the leading spaces here:
```suggestion
            (
                'msa_mlss_id',
                [
                    "UPDATE method_link_species_set_tag SET value = 1 "
                        "WHERE method_link_species_set_id = 5 AND tag = 'msa_mlss_id'",
                    "DELETE FROM method_link_species_set_tag "
                        "WHERE method_link_species_set_id = 50001 AND tag = 'msa_mlss_id'",
                    "INSERT INTO method_link_species_set_tag VALUES (404, 'msa_mlss_id', 1)"
                ],
                set([
                    "Repaired MLSS tag 'msa_mlss_id' for MLSS id '5'",
                    "Added missing MLSS tag 'msa_mlss_id' for MLSS id '50001'",
                    "Deleted unexpected MLSS tag 'msa_mlss_id' for MLSS id '404'"
                ]),
                {5: 4, 7: 6, 9: 8, 50001: 4, 50002: 6, 50003: 8},
                does_not_raise()
            ),
```

Comment (ens-sb) [reply src/python/tests/test_repair_mlss_tags.py:102]: Okay! No worries!

File: src/python/tests/test_filesys.py [hunk 5]
```diff
@@ -43,7 +44,7 @@ class BaseTestFilesys:
 
     """
 
-    dir_cmp = None  # type: DirCmp
+    dir_cmp: DirCmp = None
```
Comment (twalsh-ebi) [inline src/python/tests/test_filesys.py:47]:
```suggestion
    dir_cmp: DirCmp = None  # type: ignore
```

**Review by twalsh-ebi — APPROVED** [review]

**Pull request merged**